### PR TITLE
Add upcoming matches tab

### PR DESCRIPTION
**Pull request by kubaolszewski**

### CODE REVIEW

File: lib/app/features/home/home_tabs/upcoming_tab/cubit/upcoming_tab_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        final upcomingMatches =
+            await homeScreenRepository.fetchUpcomingMatches(AppConstVariables.matchNotStarted, formattedDate);
+        if (upcomingMatches!.isNotEmpty) {
+          emit(MatchesLoadedState(upcomingMatches.take(20).toList()));
+        }
+      } catch (error) {
+        emit(ErrorMatchesState(error.toString()));
+      }
+    } else {
+      emit(const ErrorMatchesState('No results.'));
```
Comment (MBanaszynski) [inline lib/app/features/home/home_tabs/upcoming_tab/cubit/upcoming_tab_cubit.dart]: Add to localise

File: lib/repositories/home_screen_repository.dart [hunk 2]
```diff
@@ -313,8 +308,8 @@ class HomeScreenRepository {
     return response.map(LiveMatchModel.fromJson).toList();
   }
 
-  Future<List<LiveMatchModel>?> fetchLiveMatches() async {
-    // return (await liveMatchesRemoteService.fetchLiveMatches(live: 'all')).response;
+  Future<List<LiveMatchModel>?> fetchMatchesByDate(String status, String date) async {
+    // return (await liveMatchesRemoteService.fetchMatchesByDate(status: AppConstVariables.fullTime, date: date)).response;
```
Comment (MBanaszynski) [inline lib/repositories/home_screen_repository.dart:312]: I believe that's on purpose?

Comment (kubaolszewski) [reply lib/repositories/home_screen_repository.dart:312]: if you mean commented code - yes 😅

File: lib/app/features/home/home_tabs/upcoming_tab/cubit/upcoming_tab_cubit.dart (hunk 1 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_tabs/upcoming_tab/cubit/upcoming_tab_cubit.dart]: how can I do that when I don't have context available to pass into `context.localizations...` needed by extension?

**Review by MBanaszynski — APPROVED** [review]

**Pull request merged**